WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt

```diff
@@ -250,9 +254,14 @@ class PagesViewModel
         pageMap = pageMap
     }
 
-    fun onPageEditFinished() {
+    fun onPageEditFinished(localPageId: Int, data: Intent) {
         launch {
             refreshPages() // show local changes immediately
+            withContext(defaultDispatcher) {
+                pageStore.getPageByLocalId(pageId = localPageId, site = site)?.let {
+                    _postUploadAction.postValue(Triple(it.post, it.site, data))
+                }
+            }
         }
     }
```

Comment:
I am still getting a solid idea of how the `pageStore` component workss. Since the `refreshPages` was called above could we utilize that `PagedModel` instead of making another call to the `pageStore`. Something like this 

```kotlin
fun onPageEditFinished(remotePageId: Long, data: Intent) {
        launch {
            refreshPages() // show local changes immediately
            withContext(defaultDispatcher) {
                pageMap[remotePageId]?.let {
                    _postUploadAction.postValue(Triple(it.post, it.site, data))
                }
            }
        }
    }
```
Or is there a reason we need to fetch it from the `pageStore`. Let me know if it makes sense or not 😅

Reply:
Great question! I tried the approach you suggested and right after I committed the changes I remembered why I didn't use that approach yesterday :D. LocalDrafts don't have remoteId since they weren't uploaded to the server yet. However, we want to emit the postUploadAction even for those. We could try to get the PageModel from `pageMap` with remoteId and if it's not there fetch it from database, but it seems like a super minor optimisation which isn't worth it imo. If we were loading the data from database while scrolling or something like that, it'd make total sense. But we hit the database just once on a background thread when the user closes the editor.

First attempt committed in a3bf1776a423d8b30d6e0211e180608a032f4970. Partial revert committed in f9b6a52168c8406b7292cc592eaecf81164b4621.

Reply:
> Great question! I tried the approach you suggested and right after I committed the changes I remembered why I didn't use that approach yesterday :D. LocalDrafts don't have remoteId since they weren't uploaded to the server yet.

Thanks for this! Yes, that's true. I got a `NullPointerException` when the `remoteId` was missing while working on a PR and had to switch to the `localId` 🙏